google-cloud-core/src/main/java/com/google/cloud/HttpTransportOptions.java

```diff
@@ -154,10 +155,26 @@ public void initialize(HttpRequest httpRequest) throws IOException {
         if (readTimeout >= 0) {
           httpRequest.setReadTimeout(readTimeout);
         }
+
+        HttpHeaders headers = httpRequest.getHeaders();
+        headers.set("x-goog-api-client", getXGoogApiClientHeader());
       }
     };
   }
 
+  String getXGoogApiClientHeader() {
+    return String.format(
+        "gl-java/%s %s/%s",
```

Comment:
@bjwatson So... I know that maybe I'm being a huge pain here, but I'm concerned about the lack of indication within this header that we're using the old-platform stack. There is no 'gdcl' key, no key to indicate the version of google-api-java-client, etc. This concerns me for the case of Datastore, where we intend to switch from old-platform to grpc down the road. I know that the 'gdcl' key was going to be something we added later, but my understanding was that was because we weren't adding this header for old-platform APIs until later; now, we are adding the header.